src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1017,7 +1019,86 @@ else if (block instanceof BlockLiquid)
             return eyes < pos.getY() + 1 + filled;
         }
     }
+    
+    /*
+     * Predicate version of Entity inside material method
+     */
+    public static boolean isEntityInsideOfMaterial(Entity entity, Predicate<Material> predicate)
+    {
+        if (entity.getRidingEntity() instanceof EntityBoat)
+        {
+            return false;
+        }
+        else
+        {
+            BlockPos pos = new BlockPos(entity.posX, entity.posY + entity.getEyeHeight(), entity.posZ);
+            Material material = entity.world.getBlockState(pos).getMaterial();
+            if (predicate.test(material))
+            {
+                // DEBUG
+                if (entity instanceof EntityPlayer) System.out.println("Passed predicate test");
```

Comment:
This debug statement and log should be removed, or hidden behind a public static boolean flag.